firewood/src/merkle/node.rs

```diff
@@ -299,27 +303,35 @@ impl ExtNode {
     }
 
     pub fn new(path: Vec<u8>, chd: DiskAddress, chd_encoded: Option<Vec<u8>>) -> Self {
-        ExtNode(PartialPath(path), chd, chd_encoded)
+        ExtNode {
+            path: PartialPath(path),
```

Comment:
Just a question, why not force the caller to pass in a partial-path so that it's more explicit?